homeassistant/components/splunk/__init__.py

```diff
@@ -128,56 +105,11 @@ async def splunk_event_listener(event):
                 "value": _state,
             },
         }
-        batch.append(json.dumps(payload, cls=JSONEncoder))
 
-        # Enforce only one loop is running
-        if lock.locked():
-            return
-        async with lock:
-            # Run until there are no new events to sent
-            while batch:
-                size = len(batch[0])
-                events = deque()
-                # Do Until loop to get events until maximum payload size or no more events
-                # Ensures at least 1 event is always sent even if it exceeds the size limit
-                while True:
-                    # Add first event
-                    events.append(batch.popleft())
-                    # Stop if no more events
-                    if not batch:
-                        break
-                    # Add size of next event
-                    size += len(batch[0])
-                    # Stop if next event exceeds limit
-                    if size > SPLUNK_SIZE_LIMIT:
-                        break
-                _LOGGER.info(
-                    "Sending %s of %s events to Splunk",
-                    len(events),
-                    len(events) + len(batch),
-                )
-                # Send the selected events
-                try:
-                    await hass.async_add_executor_job(
-                        event_collector._send_to_splunk,  # pylint: disable=protected-access
-                        "send-event",
-                        "\n".join(events),
-                    )
-                except (
-                    request_exceptions.HTTPError,
-                    request_exceptions.Timeout,
-                    request_exceptions.ConnectionError,
-                    request_exceptions.TooManyRedirects,
-                ) as err:
-                    _LOGGER.warning(err)
-                    # Requeue failed events
-                    batch = events + batch
-                    break
-                except json.decoder.JSONDecodeError:
-                    _LOGGER.warning("Unexpected response")
-                    # Requeue failed events
-                    batch = events + batch
-                    break
+        try:
+            await event_collector.queue(json.dumps(payload, cls=JSONEncoder), send=True)
+        except Exception as err:  # pylint: disable=broad-except
```

Comment:
We want to catch specific exceptions. What exception do we expect?

It's best if the library can raise one or more library specific exceptions that we can import and use here in the except block.

Reply:
So I wondered about this, what value does catching specific exceptions give us? I want to warn the user of any issues, but I don't need to take any different action as the library will attempt to resend the data anyway.

I can certainly break this out to be only the expected exceptions, but I figured they would all end up in _logger.warn anyway. Is there some logic I am missing?

Reply:
Our standard is to only catch specific exceptions in integrations except in the config flow where we usually add a fallback broad exception catch. If there are unexpected exceptions we let them leak. If we see any of those they should be fixed or handled specifically.

There are exceptions to this rule, but it should be what we aim for.

Reply:
Ok that makes sense. Ill add the aiohttp errors and create one specifically from the library for Splunk Errors. Appreciate your guidance @MartinHjelmare.

Reply:
This has now been implemented @MartinHjelmare .